CorsixTH/Lua/entities/humanoids/staff.lua

```diff
@@ -193,10 +193,11 @@ function Staff:checkIfWaitedTooLong()
         return
       end
 
-      -- Plays the sack sound, but maybe it's good that you hear a staff member leaving?
+      -- else: The staff member is sacked
       if staff_rise_window then
         staff_rise_window:fireStaff()
       else
+        self.hospital:removeMessage(self) -- close any queued wage rise requests
```

Comment:
Wouldn't it be better to move this line inside the `fire` function below? It already does more cleanup.

Reply:
I put it there originally. Then put it here because this function leans on the scope of handling the staff rise window. We also call it here for granting the raise by policy L191)
Happy to move it back if you think it's better.

Reply:
Oh, don't know, it seemed a general thing that would be needed no matter how a staff got fired, but if you think here is better, that's ok.

I don't know enough of the details of the code to give a definitive answer.

Reply:
Yeah. The die function also really should look for this if the room explodes.